src/ingestor.js

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { timeoutSignal, fetch as originalFetch } from '@adobe/fetch';
+import { forEachLimit } from 'async';
+import fetchBuilder from 'fetch-retry';
+
+// eslint-disable-next-line no-unused-vars
+import * as extractors from './extractors.js';
+
+const fetch = fetchBuilder(originalFetch);
+
+/**
+ * @typedef IngestorConfig the confiuration for the ingestor client
+ * @property {string} url the URL for calling the ingestor
+ * @property {string} apiKey the API Key used to call the ingestor
+ * @property {string} jobId the id of the current job
+ */
+
+/**
+ * The ingestor client sends asset data to the Content Lake ingestion service to be ingested
+ */
+export class IngestorClient {
+  #config;
+
+  #log;
```

Comment:
Cool, I didn't know these were officially supported yet. I remember we had to hack our own way using Symbols: https://github.com/adobe/asset-compute-pipeline/blob/master/lib/engine.js#L44